src/test/java/org/springframework/cloud/bindings/boot/EurekaBindingsPropertiesProcessorTest.java

```diff
@@ -67,5 +199,4 @@ void disabled() {
 
         assertThat(properties).isEmpty();
     }
-
```

Comment:
please add some tests that make the code throw exceptions.
What would happen? kill the whole app startup? or continue without TLS config?

Reply:
Do you have suggestions on where that test code should live? To me, this unit test maybe isn't it. We've converted everything to a runtime exception, but maybe to your point, I am not sure what the calling code will do when it gets one of our IllegalStateExceptions.

My thought is that we probably *do* want the app to fail startup if we have a binding we cannot process, but I don't think we can test whether or not that will happen in this class.

Reply:
ok, let me rephrase my suggestion.

I would like at least 1 test that fails to create a store and throws one of those `IllegalStateException`.

This is part of your changes, so it definitely makes sense to add it as part of your tests - for example, re do the test  named `@DisplayName("contributes mTLS properties when set")`  but this time with a `cert` and /or `key` like:

```java
            this.cert = "jklkl";
            this.key = "lkjlkjkl";
```

You'll be able to assert that `IllegalException` are supposed to be thrown in those cases.

---
> My thought is that we probably do want the app to fail startup if we have a binding we cannot process, but I don't think we can test whether or not that will happen in this class.

About integrations tests, we have [some](https://github.com/spring-cloud/spring-cloud-bindings/pull/94/files#diff-5fdfc03b2902a383ceb5db7636ec92877c04b81f29539c553ef92c15f4716003R10) that exist in a PR that can't be merged yet; it could be interesting to add an `Eureka` integration test that would break the startup of the app.